### ISSUE TITLE
[Schema streaming mode] dotProduct function is ignored in itemRawScore calulation

### ISSUE THREAD

**Issue by akolhun** [vespa-engine/vespa#29493]
**Describe the bug**
`dotProduct` function is ignored in doc score calulation 

**To Reproduce**
Given schema mode ="streaming" and is defined as 
```
schema contact {
    document contact {
       field test_field type weightedset<string>  {
            indexing: summary | index
            weightedset {
                create-if-nonexistent
                remove-if-zero
            }
            match: exact
            rank: normal
        }
    }
    
    rank-profile contact_default inherits default {

        first-phase {
            expression {
                itemRawScore(test_field)
            }
        }
    }
}
```
And document indexed as :
```
{
    "fields": {
        "test_field": {"computer": 1000, "compute": 800}
    }
}
```
Then the following search query returns the doc with relevance=0:
```
   "yql": "select * from contact where [{'label': 'test_field'}] dotProduct(test_field, {'computer': 1})",
    "ranking.profile": "contact_default",
```   

**Expected behavior**
The doc above should be returned with relevance=1000 (as it behaves in "index" mode)

**Environment (please complete the following information):**
docker image: vespaengine/vespa:8.263.7

**Vespa version**
8.263.7

**Additional context**
Issues might be reproduced within the app package attached.
[vespa-streaming.zip](https://github.com/vespa-engine/vespa/files/13500571/vespa-streaming.zip)
Indexing request:
```
curl --location 'http://localhost:8080/document/v1/mp/contact/group/test_group/docid/555561' \
--header 'Content-Type: application/json' \
--data '{
    "fields": {
        "test_field": {"computer": 1000, "compute": 800}
    }
}
'
```
Search request:
```
curl --location 'http://localhost:8080/search/' \
--header 'Content-Type: application/json' \
--data '{
    "user": "ak",
     "yql": "select * from contact where [{'\''label'\'': '\''test_field'\''}] dotProduct(test_field, {'\''computer'\'': 1})",
     "ranking.profile": "contact_default",
     "streaming.groupname": "test_group"

}'
```

**Comment by akolhun** [comment]
@kkraune there was also reported  https://github.com/vespa-engine/vespa/issues/29514 in the context of "streaming search". It might be related. thx

**Comment by jobergum** [comment]
Please consider using tensors for calculating dot products as a work-around for this issue. See for example https://docs.vespa.ai/en/tensor-examples.html#dot-product-between-query-and-document-vectors

**Comment by akolhun** [comment]
@jobergum thx, the suggested approach works, but looks like it's limited to bounded tensors only.
In our case, `dotProduct`  should be applied to:
```
query(query_vector)  =  {{x:5}: 2}
document_vector = {{x:1}: 300, {x:5}: 1000}}
```
As a result, dotProduct should result into score/relevance = 2000.

Tried out the definition below, but it's not working.
Am I missing anything here?


```
schema example {
    document example {
        field document_vector type tensor<float>(x{}) {
            indexing: attribute | summary 
        }

    }

      rank-profile dot_product {
        inputs {
            query(query_vector) tensor<float>(x{})
        }
        first-phase {
            expression: sum(query(query_vector)*attribute(document_vector))
        }
    }

}
```

**Comment by jobergum** [comment]
Not sure what you mean by bound tensors, but you can perform the sparse or dense dot product. Here using a mapped tensor. 

```
schema mail {
    document mail {
        field features type tensor<float>(x{}) {
            indexing: summary | attribute
        }
    }
    rank-profile dot {
      inputs {
        query(q_features) tensor<float>(x{})
      }
      first-phase {
        expression: sum(query(q_features) * attribute(features))
      }
    }
}
```

### feed

```json { "id": "id:mail:mail:g=bergum@vespa.ai:0", "fields": { "features": { "foo":1}}}```

### query 

```vespa query 'yql=select * from sources mail where true' 'streaming.groupname=bergum@vespa.ai' 'input.query(q_features)={foo:2}' 'ranking=dot' 'presentation.format.tensors=short-value'   
{
    "root": {
        "id": "toplevel",
        "relevance": 1.0,
        "fields": {
            "totalCount": 1
        },
        "coverage": {
            "coverage": 100,
            "documents": 1,
            "full": true,
            "nodes": 1,
            "results": 1,
            "resultsFull": 1
        },
        "children": [
            {
                "id": "id:mail:mail:g=bergum@vespa.ai:0",
                "relevance": 2.0,
                "source": "assistant_content.mail",
                "fields": {
                    "sddocname": "mail",
                    "documentid": "id:mail:mail:g=bergum@vespa.ai:0",
                    "features": {
                        "foo": 1.0
                    }
                }
            }
        ]
    }
}
```

**Comment by akolhun** [comment]
Confirmed. it works. Thank you!

Note: it was not working on my end due to    `ranking.listFeatures=true` parameter.
Try out:
```
vespa query 'yql=select * from  mail where true' 'streaming.groupname=test_group' 'ranking.features.query(q_features)={"foo": 2}' 'ranking.profile=dot' 'ranking.listFeatures=true'

{
    "root": {
        "id": "toplevel",
        "relevance": 1.0,
        "fields": {
            "totalCount": 0
        },
        "coverage": {
            "coverage": 100,
            "documents": 1,
            "full": true,
            "nodes": 1,
            "results": 1,
            "resultsFull": 1
        }
    }
}
```
while `ranking.listFeatures=false` (or skipped)  returns the response above

**Comment by jobergum** [comment]
Thanks, created https://github.com/vespa-engine/vespa/issues/29588 on that.

**Comment by geirst** [comment]
The fix will be part of Vespa 8.275.39.
